### DISCUSSION TITLE
Transform Layer is constructed with an invalid matrix

### DISCUSSION THREAD

**Discussion by 7576457** [flet-dev/flet#1855] [Q&A] [answered]
### Question

When I add an instance of the Sec class to the page, I get this error.  Why is this happening and how can it be fixed?

### Code sample

```python
from flet import (
    Row,
    Column,
    Container,
    Text,
    UserControl,
    border_radius,
    colors,
    icons,
    Icon,
    IconButton,
    ImageFit,
    alignment,
    Tabs, Tab,
    AlertDialog,
    TextField,
    InputBorder,
    TextButton,
    MainAxisAlignment,
    Page,
    app
)

class Sec(UserControl):
    def build(self):

        self.tabs = Tabs(
            animation_duration=300,
            selected_index=0,
            tabs=[
                Tab(text="STORAGE", content=Container(bgcolor='red')),
                Tab(text="FILE", content=Container(bgcolor='blue')),
            ],
            expand=1,
        )
        
        return self.tabs
    

def main(page: Page):

    tol = Sec()
    page.add(tol)

app(target=main)
```

### Error message

```bash
[ERROR:flutter/flow/layers/transform_layer.cc(24)] TransformLayer is constructed with an invalid matrix.
[ERROR:flutter/flow/layers/transform_layer.cc(24)] TransformLayer is constructed with an invalid matrix.
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by FeodorFitsner** [comment]
What OS is that?

**Reply by 7576457** [reply]
Linux Ubuntu 22.04 LTS Gnome

**Reply by FeodorFitsner** [reply]
The latest Flet is running fine on Ubuntu 22.04. But I guess I'm running non-Gnome window manager, right?

![image](https://github.com/flet-dev/flet/assets/5041459/83023e97-d3be-4251-b801-078e0c313304)

**Reply by 7576457** [reply]
In your photo, at least visually, everything looks the same as in mine. This example, which is shown in your screenshot, works fine on my system. 

I will add that I have a window manager that is included when installing Ubuntu - mutter. The window interface is Wayland.

I will also supplement my question: I realized that when adding the "content" parameter to an object of the Tab class, it throws the presented error.  At the same time, if I transfer the code from the Sec class to the 'main' function, then it will work calmly

**Reply by FeodorFitsner** [reply]
Indeed, I can reproduce the issue with tabs on Ubuntu:

<img width="1272" alt="image" src="https://github.com/flet-dev/flet/assets/5041459/0610518f-9d28-414f-8f71-1e725433a682">

**Reply by FeodorFitsner** [reply, marked as answer]
OK, found the issue! User control itself must be expanded (`self.expand = True`):

```python
from flet import (
    Container,
    UserControl,
    Tabs, Tab,
    Page,
    app
)

class Sec(UserControl):
    def build(self):

        self.expand = True
        self.tabs = Tabs(
            animation_duration=300,
            selected_index=0,
            tabs=[
                Tab(text="STORAGE", content=Container(bgcolor='red')),
                Tab(text="FILE", content=Container(bgcolor='blue')),
            ],
            expand=1,
        )
        
        return self.tabs
    

def main(page: Page):

    tol = Sec()
    page.add(tol)

app(target=main)
```
<img width="630" alt="image" src="https://github.com/flet-dev/flet/assets/5041459/8be23df0-de4a-41c5-b926-79befd7d1e55">

I have to admit the error is misleading.
